File: algorithms/sort/mergeSort/Merge.java
package algorithms.sort.mergeSort;

import java.util.Random;
import java.util.Arrays;

public class Merge {
    public static void setMerge(int[] arr) {
        int arrLength = arr.length;
        if (arrLength > 1) {
            
            int mid = arrLength / 2;
            
            int[] leftArr = new int[mid];
            int[] rightArr = new int[arrLength-mid];
                
            System.arraycopy(arr, 0, leftArr, 0, leftArr.length);
            System.arraycopy(arr, mid, rightArr, 0, rightArr.length);

            setMerge(leftArr);
            setMerge(rightArr);

            int i = 0;
            int j = 0;
            int k = 0;

            while (i<leftArr.length && j<rightArr.length) {
                if (leftArr[i] < rightArr[j]) {
                    arr[k] = leftArr[i];
                    i++;
                }else{
                    arr[k] = rightArr[j];
                    j++;
                }
                k++;
            }

            while (i<leftArr.length) {
                arr[k] = leftArr[i];
                i++;
                k++;
            }

            while (j<rightArr.length) {
                arr[k] = rightArr[j];
                j++;
                k++;
            }
        }
    }

    public static void main(String[] args) {
        int listLength = 27;
        int[] originalList = new int[listLength];
        Random random = new Random();

        for (int i=0; i<originalList.length; i++) {
            int randNum = random.nextInt(100); 
            originalList[i] = randNum;
        }
        setMerge(originalList);
        System.out.println(Arrays.toString(originalList));
    }
}

File: sort/mergeSort/merge.py
import random

def set_merge_sort(arr):
    arr_size = len(arr)
    if arr_size > 1:
        mid = arr_size // 2  # 중간 인덱스 계산
        left_arr = arr[:mid]  # 왼쪽 부분 배열
        right_arr = arr[mid:]  # 오른쪽 부분 배열

        # 왼쪽 부분 배열과 오른쪽 부분 배열을 재귀적으로 정렬
        set_merge_sort(left_arr)
        set_merge_sort(right_arr)

        i = j = k = 0   
            
        # 두 부분 배열을 합치기
        while i < len(left_arr) and j < len(right_arr):
            if left_arr[i] < right_arr[j]:
                arr[k] = left_arr[i]
                i += 1
            else:
                arr[k] = right_arr[j]
                j += 1
            k += 1

        # 남은 요소들 처리
        while i < len(left_arr):
            arr[k] = left_arr[i]
            i += 1
            k += 1

        while j < len(right_arr):
            arr[k] = right_arr[j]
            j += 1
            k += 1

if __name__ == "__main__":
    original_list = [random.randint(0,100) for _ in range(10)]

    # 병합 정렬 수행
    set_merge_sort(original_list)

    # 정렬된 결과 출력
    print("정렬된 리스트:", original_list)